### ISSUE TITLE
Two issues regarding the implementation

### ISSUE THREAD

**Issue by hubert0527** [shaanrockz/cocogan#1]
Thanks for your interest in COCO-GAN and your effort in implementing it in Pytorch!

I took a quick look and found two issues (not sure if there is still any other) that may cause your model failed to converge:

1. The implementation of CBN is wrong, the `mean_rec` and `var_rec` needs to be tracked for each iteration. It should work very similarly to the standard batch norm. You may refer to this:
https://discuss.pytorch.org/t/conditional-batch-normalization/14412/2

2. You will have to implement a GPU version for [micro-patch-to-macro-patch conversion](https://github.com/shaanrockz/COCO-GAN/blob/09e33b6912e224f292f0bd0231c1af611840bad6/trainer.py#L170). Otherwise, there will be no gradient propagation from the discriminator for the generator adversarial training.

**Comment by shaanrockz** [comment]
Thank you for pointing out, appreciated. Will update the code.
The code was debugged and run on the cpu for implementation, I should have added this in the description, my bad. Anyways I will also add the gpu implementation of the functions soon.

**Comment by waydegilliam** [comment]
Have these issues been fixed yet?

**Comment by shaanrockz** [comment]
> Have these issues been fixed yet?

Hi, sorry for late reply. I will update the codebase soon.
